lib/active_model/serializer/adapter/attributes.rb

```diff
@@ -46,10 +62,59 @@ def fragment_cache(cached_hash, non_cached_hash)
 
         private
 
+        def relevant_included_associations_for(serializer, nested_associations = {})
```

Comment:
you don't need to pass serializer in. It's already available.  And all these methods using it as an argument are probably a latent object. https://github.com/kerrizor/latent_object_detector